cdk/lib/auth/sso-auth-stack.ts

```diff
@@ -0,0 +1,137 @@
+import { FederatedPrincipal, PolicyStatement, Role } from 'aws-cdk-lib/aws-iam';
+import { Construct } from 'constructs';
+
+export interface SsoAuthStackProps extends StackProps {
+  readonly applicationName: string;
+  readonly logGroupArn: string;
+  readonly removalPolicyOverride?: RemovalPolicy;
+}
+
+export class SsoAuthStack extends Stack {
+  readonly identityPool: CfnIdentityPool;
+  readonly userPool: UserPool;
+  readonly userPoolClient: UserPoolClient;
+  readonly domain: UserPoolDomain;
+
+  constructor(scope: Construct, id: string, props: SsoAuthStackProps) {
+    super(scope, id, props);
+
+    const { applicationName, logGroupArn, removalPolicyOverride } = props;
+
+    this.userPool = new UserPool(this, 'UserPool', {
+      signInCaseSensitive: false,
+      removalPolicy: removalPolicyOverride,
+    });
+
+    this.userPoolClient = new UserPoolClient(this, 'UserPoolClient', {
+      userPool: this.userPool,
+      oAuth: {
+        flows: {
+          implicitCodeGrant: true,
+        },
+        callbackUrls: ['https://<your-url>.awsapprunner.com'],
+      },
+    });
+
+    this.domain = this.userPool.addDomain('Domain', {
+      cognitoDomain: {
+        // Use the unique id for this cdk construct for naming
+        domainPrefix: `sitewise-${this.node.addr.substring(0, 6)}`,
+      },
+    });
+
+    this.domain.signInUrl(this.userPoolClient, {
+      redirectUri: 'https://<your-url>.awsapprunner.com',
+    });
+
+    this.identityPool = new CfnIdentityPool(this, 'IdentityPool', {
+      allowUnauthenticatedIdentities: false,
+      cognitoIdentityProviders: [
+        {
+          clientId: this.userPoolClient.userPoolClientId,
+          providerName: this.userPool.userPoolProviderName,
+        },
+      ],
+    });
+
+    const authenticatedRole = new Role(this, 'AuthenticatedRole', {
+      assumedBy: new FederatedPrincipal(
+        'cognito-identity.amazonaws.com',
+        {
+          StringEquals: {
+            'cognito-identity.amazonaws.com:aud': this.identityPool.ref,
+          },
+          'ForAnyValue:StringLike': {
+            'cognito-identity.amazonaws.com:amr': 'authenticated',
+          },
+        },
+        'sts:AssumeRoleWithWebIdentity',
+      ),
+    });
+
+    // IoT CloudWatch Access
+    authenticatedRole.addToPolicy(
+      new PolicyStatement({
+        actions: ['cloudwatch:PutMetricData'],
+        resources: ['*'],
+        conditions: {
+          StringEquals: {
+            'cloudwatch:namespace': applicationName,
+          },
+        },
+      }),
+    );
+
+    // IoT CloudWatch Logs Access
+    authenticatedRole.addToPolicy(
+      new PolicyStatement({
+        actions: ['logs:CreateLogStream', 'logs:PutLogEvents'],
+        resources: [logGroupArn],
+      }),
+    );
+
+    // IoT Data Access
+    authenticatedRole.addToPolicy(
+      new PolicyStatement({
+        actions: [
+          'iotevents:DescribeAlarmModel',
+          'iotevents:ListTagsForResource',
```

Comment:
> [!NOTE]  
nit: we will forgot to update the permission policy in the future
Would be better to have a base stack/class for common resources between the auth stacks
Even better would be providing a common interface for inversion